napari/_vispy/vispy_image_layer.py

```diff
@@ -66,14 +66,17 @@ def _data_astype(self, data, dtype):
         return data.astype(dtype)
 
     def _on_data_change(self, event=None):
-        """Our self.layer._data_view has been updated, update our node.
-        """
         if not self.layer.loaded:
             # Do nothing if we are not yet loaded. Calling astype below could
             # be very expensive. Lets not do it until our data has been loaded.
             return
 
-        data = self.layer._data_view
+        self._set_node_data(self.node, self.layer._data_view)
+
+    def _set_node_data(self, node, data):
+        """Our self.layer._data_view has been updated, update our node.
+        """
```

Comment:
I'm fine with this, is very clever for usage in tiled version

Reply:
This works well since we are letting`VispyImageLayer` do all the pre-processing it was always doing. The derived `VispyTiledImageLayer` is just calling into `VispyImageLayer._set_node_data()` once per tile. So yeah this was nice.

If we do switch to a `TiledImageVisual` as David suggested, I'm not as positive who would do this preprocessing. Maybe `VispyImageLayer` can continue to do it, and it just sends the processed data to the `TiledImageVisual` rather than making an `ImageNode` out of it, like we do here. That would also be nice.

Reply:
> If we do switch to a TiledImageVisual as David suggested, I'm not as positive who would do this preprocessing. Maybe VispyImageLayer can continue to do it, and it just sends the processed data to the TiledImageVisual rather than making an ImageNode out of it, like we do here. That would also be nice.

Indeed I was wondering this too, but we can address later